dvc/repo/get.py

```diff
@@ -17,6 +19,30 @@
 logger = logging.getLogger(__name__)
 
 
+def _is_git_file(repo, path):
+    if not os.path.isabs(path):
+        try:
+            output = repo.find_out_by_relpath(path)
```

Comment:
Feels weird that you need to lookup output twice. Once here and once in the original get() code.

Reply:
I did not find a cleaner approach consider that handling of the error varies in the two cases of git vs dvc file.

Reply:
This is why we handle cached outs first in Repo.open()

Reply:
@danihodovic Have you seen `dvc/repo/__init__.py::open()`?

Reply:
I have, but I can't get similar logic to fit into this context because of an outer catch clause of `OutputNotFoundError`.

If you have a better solution I would love to see it.

Reply:
@danihodovic So why not move it where you need it? Sorry, might be missing something.

Reply:
Can you elaborate? Do you want me to move the [is git logic](https://github.com/danihodovic/dvc/blob/feat/2515/dvc/repo/get.py#L23-L31) to `dvc/repo/__init__.py`?

Reply:
@danihodovic You were saying that 

> I can't get similar logic to fit into this context because of an outer catch clause of OutputNotFoundError.

So I wonder why you can't adjust it to have similar logic so that you don't lookup output twice in your code? I.e. you have find_outs_by_relpath here https://github.com/iterative/dvc/pull/2837/files#diff-d84a7ca40e278c5be8904f9aeb922da0R26 and then here https://github.com/iterative/dvc/pull/2837/files#diff-d84a7ca40e278c5be8904f9aeb922da0R83 . Similar code in Repo.open() has only one find_outs_by_path and is dealing with absolutely the same situation where it has to distinguish cached outs from non-cached outs from git files. So I'm wondering why you can't do something similar here in get()?

Reply:
Do you find this to be a cleaner solution? 
https://github.com/iterative/dvc/compare/master...danihodovic:feat/2515-alternative?expand=1#diff-d84a7ca40e278c5be8904f9aeb922da0R68-R77